sqlglot/generator.py

```diff
@@ -218,6 +219,15 @@ def sep(self, sep=" "):
     def seg(self, sql, sep=" "):
         return f"{self.sep(sep)}{sql}"
 
+    def maybe_comment(self, sql, comment):
+        if not comment or not self._comments:
+            return sql
+
+        if not self.pretty or "\n" not in comment:
+            return f"{sql} /* {comment.strip()} */"
```

Comment:
i think we never want to strip also your newline check isn't cross platform, you may need to do a newline check 

https://stackoverflow.com/questions/1331815/regular-expression-to-match-cross-platform-newline-characters

Reply:
The reason I call strip here is that it's awkward to have `/* text*/`, whenever the input sql has a `--` comment. I guess it does make sense to never strip comments, if for example you want to do something like the following in SQL:

```sql
-- Args:
--     foo: ...
```
Although, at the time of writing this I was thinking what I wrote in [this](https://github.com/tobymao/sqlglot/pull/663#discussion_r1012459775) reply. I can do it so that we never strip if you think that's the correct way.